cmd/cli/plugin/airgapped-network/copyimage.go

```diff
@@ -0,0 +1,229 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	tkrv1 "github.com/vmware-tanzu/tanzu-framework/apis/run/pkg/tkr/v1"
+)
+
+type imageCopyOptions struct {
+	customImageRepo                 string
+	tkgImageRepo                    string
+	tkgVersion                      string
+	customImageRepoCertificate      string
+	customTkgCompatibilityImagePath string
+	ignoreCompatibilityImage        bool
+}
+
+var totalImgCopiedCounter int
+var lc = &imageCopyOptions{}
+
+var imageCopyCmd = &cobra.Command{
+	Use:          "copyimage",
+	Short:        "Copy images to airgapped network",
+	RunE:         copyimages,
+	SilenceUsage: true,
+}
+
+func init() {
+	imageCopyCmd.Flags().StringVarP(&lc.customImageRepo, "custom-image-repo", "", "", "custom images repository for airgapped network")
+	imageCopyCmd.Flags().StringVarP(&lc.tkgImageRepo, "tkg-image-repo", "", "", "TKG public repository")
+	imageCopyCmd.Flags().StringVarP(&lc.tkgVersion, "tkg-version", "", "", "TKG version")
+	imageCopyCmd.Flags().StringVarP(&lc.customImageRepoCertificate, "custom-imageRepo-certificate", "", "", "customImageRepoCertificate")
+}
+
+func handleTkgCompatibilityImage() {
+	fmt.Printf("--- start the process of tkg-compatibility ---")
+	tkgCompatibilityRelativeImagePath := "tkg-compatibility"
+	if !IsTagRTM(lc.tkgVersion) {
+		tkgCompatibilityRelativeImagePath = filepath.Join(lc.tkgVersion, tkgCompatibilityRelativeImagePath)
```

Comment:
This may not be a valid one.

Reply:
@sathyanarays  Please can you explain in detail ?